cms/src/api/habitat-stat/controllers/habitat-stat.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * habitat-stat controller
+ */
+
+import { factories } from '@strapi/strapi'
+
+export default factories.createCoreController('api::habitat-stat.habitat-stat', ({ strapi }) => ({
+    async find(ctx) {
+        // find the most recently updated record and return its updatedAt date
```

Comment:
Thinking about it, I don't agree - I would do the same if it was a Rails CRUD, since I'm not modifying what the action does, just adding metadata related to the result. Actually I would also overwrite completely what a CRUD action does as long as the end result is consistent with the semantics and response format of the action.

What worries me more here is that this is probably going to be needed for other endpoints as well and so this will be repeated in multiple controllers :hankey: . The controller index actions for resources called "something-something-stats" are designed to return pre-aggregated data for a particular chart, and we'll need this last update information in every case. So I think what I might do here to make this less in the face is move the logic around getting the timestamp to a service, especially that I think it might change and won't need to be taken from the returned objects, this way the change to controller action will be minimal.

Reply:
oh, sorry, misread your code, and thought it did something else. carry on